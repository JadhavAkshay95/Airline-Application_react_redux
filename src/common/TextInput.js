import React from 'react'
import PropTypes from 'prop-types'

const TextInput = ({
  name,
  label,
  handleChange,
  placeholder,
  value,
  required
}) => {
  const wraperClass = 'form-group'
  return (
    <div className={wraperClass} style={{ textAlign: 'initial' }}>
      <label htmlFor={name}>{label}</label>
      {required ? <label htmlFor={name}>*</label> : null}

      <div className="field">
        <input
          type="text"
          placeholder={placeholder}
          value={value || ''}
          name={name}
          onChange={handleChange}
          className="form-control"
        />
      </div>
    </div>
  )
}

export default TextInput
TextInput.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  handleChange: PropTypes.func,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  required: PropTypes.bool
}
