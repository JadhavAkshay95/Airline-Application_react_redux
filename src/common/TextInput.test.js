import React from 'react'
import { shallow } from 'enzyme'
import TextInput from './TextInput'

const renderTextInput = args => {
  const defaultProps = {
    name: null,
    label: null,
    placeholder: null,
    value: null,
    required: null,
    handleChange: () => {}
  }
  const props = { ...defaultProps, ...args }
  return shallow(<TextInput {...props} />)
}

describe('Render text input', () => {
  it('Render text input', () => {
    const wrapper = renderTextInput({
      name: 'akshay',
      lable: 'name',
      placeholder: 'name',
      value: 'akshay',
      required: true
    })
    expect(wrapper).toMatchSnapshot()
  })
})
