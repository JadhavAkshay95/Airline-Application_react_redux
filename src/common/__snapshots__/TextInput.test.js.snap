// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render text input Render text input 1`] = `ShallowWrapper {}`;
